proxy/http/HttpTransact.cc

```diff
@@ -3692,7 +3692,7 @@ HttpTransact::handle_response_from_parent(State *s)
       s->current.attempts++;
 
       // Are we done with this particular parent?
-      if ((s->current.attempts - 1) % s->txn_conf->per_parent_connect_attempts != 0) {
+      if (s->current.attempts % s->txn_conf->per_parent_connect_attempts != 0) {
```

Comment:
Seems to match ATS 10:
https://github.com/apache/trafficserver/blob/f180ea2b6f25f6922c881b60d70cef68db1acac5/src/proxy/http/HttpTransact.cc#L3617